winsup/cygwin/dcrt0.cc

```diff
@@ -1360,7 +1360,10 @@ multiple_cygwin_problem (const char *what, uintptr_t magic_version, uintptr_t ve
     system_printf ("%s magic number mismatch detected - %p/%ly", what, magic_version, version);
   else
     api_fatal ("%s mismatch detected - %ly/%ly.\n\
-This problem is probably due to using incompatible versions of the cygwin DLL.\n\
+This problem is probably due to using incompatible versions of the msys2 or cygwin DLL.\n\
+msys2 is uses pinned memory to share child information. All versions running must be the same magic_version.\n\
+The magic versions are seen above as magic_version/version.\n\
+If cygwin1.dll is installed then make sure there is only one version installed.\n\
```

Comment:
Since you rightfully point out that there might be an MSYS2 runtime somewhere, let's mention `msys-2.0.dll` here, too.

Reply:
I didn't mention it here, because I think the advice is bad, but I did add another section underneath with better advice. Still not great, but at least it creates clarity.